app/controllers/application_controller.rb

```diff
@@ -1,7 +1,25 @@
 class ApplicationController < ActionController::Base
   rescue_from ActiveRecord::RecordNotFound, with: :render_not_found
 
+  layout :set_layout
+
+  before_action :set_variant
+
+  private
+
   def render_not_found
     render file: "#{Rails.root}/public/404.html", layout: false, status: :not_found
   end
+
+  def set_variant
+    request.variant = :mobile if mobile?
+  end
+
+  def set_layout
+    mobile? ? 'mobile' : 'desktop'
+  end
+
+  def mobile?
+    request.user_agent =~ /Mobile|webOS/
```

Comment:
Maybe there is a better way / a more complete regex to distinguish mobile from desktop browsers.

Reply:
It seems like it is bad practice to deliver different html based on the type of browser of the client. See the [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent) for an explanation.

Reply:
Instead of serving the variant according to the user-agent string, we could let the user decide which variant they want and only suggest switching to the mobile variant if a mobile browser is detected or the screen has a small resolution.
This could be saved in the session cookie or the mobile variant could be served on a subdomain (the better, but more complicated version, in my opinion).

Reply:
I think we can leave it like is and switch to a more complicated approach when problems arise.